be/src/storage/merge_iterator.cpp

```diff
@@ -64,7 +64,23 @@ class ComparableChunk : public MergingChunk {
     // assume both |this| and |rhs| are not empty.
     bool less_than_all(const ComparableChunk& rhs) {
         size_t last_row = _chunk->num_rows() - 1;
-        int r = compare_chunk(_key_columns, *_chunk, last_row, *rhs._chunk, rhs._compared_row);
+        return less_than(last_row, rhs);
+    }
+
+    // return the last row whose key value is less than all values in |rhs|
+    size_t last_row_less_than(const ComparableChunk& rhs, size_t limit_num) {
+        // As we previously pop this chunk from the heap top, `_compared_row` in this chunk
+        // must be less than all rows in rhs, thus here we start comparision from _compared_row + 1;
+        size_t last_row = _compared_row + 1;
+        size_t upper_bound = std::min(_compared_row + limit_num, _chunk->num_rows() - 1);
+        while (last_row <= upper_bound && less_than(last_row, rhs)) {
```

Comment:
why not use binary search?

Reply:
If all elements are random, min_chunk contains only a few rows (may be less than 10) whose key values are less than all other chunks. In this case, binary search is slower than linear search.